zmon-controller-app/src/main/java/de/zalando/zmon/rest/CheckDefinitionsApi.java

```diff
@@ -39,6 +39,14 @@ public CheckDefinition createOrUpdate(@Valid
         return zMonService.createOrUpdateCheckDefinition(checkDefinition);
     }
 
+    @RequestMapping(method = RequestMethod.DELETE)
+    @ResponseStatus(HttpStatus.OK)
+    public void deleteCheckDef(@Valid @RequestBody(required = true)
+            final CheckDefinitionImport checkDefinition) throws ZMonException {
```

Comment:
TL;DR: it was easier to implement this way ;-)

If it only takes an ID, then I'm able to remove any check definition, which I would personally find error-prone. So I should be able only to remove stuff, for which I'm a member of the owning team. I'm not sure how this is checked right now for the creation of check definitions? It should be consistent with that.

By accepting a body you're consistent in the command line:

```
zmon check-definitions update check.yaml
zmon check-definitions delete check.yaml
```

but you're right, the cmd line client could also check the rights and then use an ID only.